exporters/write_buffer.py

```diff
@@ -231,10 +230,13 @@ def get_key_from_item(self, item):
     def grouping_info(self):
         return self.items_group_files.grouping_info
 
-    def get_metadata(self, buffer_path, meta_key):
-        return self.metadata.get(buffer_path, {}).get(meta_key)
+    def get_metadata(self, key, module='write_buffer'):
+        return super(WriteBuffer, self).get_metadata(key, module)
+
+    def get_all_metadata(self, module='write_buffer'):
+        return super(WriteBuffer, self).get_all_metadata(module)
 
     def set_metadata_for_file(self, file_name, **kwargs):
-        if file_name not in self.metadata:
-            self.metadata[file_name] = {}
-        self.metadata[file_name].update(**kwargs)
+        if file_name not in self.get_all_metadata():
+            self.set_metadata(file_name, {})
```

Comment:
Missing method here: you have to override set_metadata() method too, as super one needs an extra (module) parameter.